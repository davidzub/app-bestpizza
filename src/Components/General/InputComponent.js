import React from "react";
import "./styles.css";

const InputComponent = ({
  placeholder,
  icon = null,
  type = "text",
  disabled,
  onChange,
}) => {
  const backIcon =
    {
      backgroundImage: `url(${icon})`,
    } || {};
  return (
    <input
      type={type}
      placeholder={placeholder}
      style={backIcon}
      className="inputLogin"
      disabled={disabled}
      onChange={(e) => onChange(e.target.value)}
    />
  );
};

export default InputComponent;
