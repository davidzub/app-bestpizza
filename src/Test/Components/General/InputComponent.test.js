import React from "react";
import { shallow } from "enzyme";
import InputComponent from "../../../Components/General/InputComponent";
describe("<InputComponent />", () => {
  let wrapper;
  const props = {
    placeholder:"",
    icon:"",
    type:"",
    disabled: false,
    onChange: jest.fn()
  }
  beforeEach(() => {
    wrapper = shallow(<InputComponent {...props}/>)
  });
  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("input change", () => {
    const event = {
        preventDefault() {},
        target: { value: 'the-value' }
      };
      const input = wrapper.find('input')
      input.simulate('change', event);
      expect(props.onChange).toBeCalledWith('the-value');
  });
});
