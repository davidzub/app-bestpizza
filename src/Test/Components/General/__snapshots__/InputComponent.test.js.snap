// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent /> render 1`] = `
<input
  className="inputLogin"
  disabled={false}
  onChange={[Function]}
  placeholder=""
  style={
    Object {
      "backgroundImage": "url()",
    }
  }
  type=""
/>
`;
